uint8 id
int32 velocity
---
int32 result
